spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/messaging/MessagingSpanExtractor.java

```diff
@@ -33,29 +34,32 @@
 public class MessagingSpanExtractor implements SpanExtractor<Message> {
 
 	private final Random random;
+	private final TraceHeaders traceHeaders;
 
-	public MessagingSpanExtractor(Random random) {
+	public MessagingSpanExtractor(Random random, TraceHeaders traceHeaders) {
 		this.random = random;
+		this.traceHeaders = traceHeaders;
 	}
 
 	@Override 
 	public Span joinTrace(Message carrier) {
-		if (!hasHeader(carrier, Span.TRACE_ID_NAME)
-				|| !hasHeader(carrier, Span.SPAN_ID_NAME)) {
+		if (!hasHeader(carrier, this.traceHeaders.getTraceId())
```

Comment:
This is fragile. This assumes that a change to how inject/extract work will only include the header name (and not the encoding of the value). Thats an odd special case that costs instructure to support.

The type injector/extractor are made precisely to support concrete changes to propagation headers, for example where values are encoded differently or bundled all into a single header.

If I were you, I would do this:

First, is to change the names and rename the injector/joiner to Zipkin (or B3).
Second, wait for a concrete use case before adding customization, as injector/extractor are already possible to adjust.